### PR TITLE
logtracing: export span data

### PR DESCRIPTION
**Pull request by w11th**
- Add the ability to export span data
- Add a Honeycomb exporter

With the changes, we can use `logtracing` in the cloud functions.

### CODE REVIEW

**Comment by w11th** [comment]
@bodhi I'm trying to add a feature to export span data. With this, we can use `logtracing` + `Honeycomb exporter` in CFs. It's also a preparation for remote tracing.

Could you please review it when possible?

**Review by bodhi — CHANGES_REQUESTED** [review]
@w11th can you please update the README for log tracing to show
1. How to set this up (vs using the standard logging output)
2. What (if anything) changes on the client side when using this exporter vs the logging exporter.

Also, at what point should we switch to OTEL, with our own exporter?

**Comment by w11th** [comment]
Thanks, @bodhi. I have updated the README to add a paragraph "Export span data": [9a30936](https://github.com/theplant/appkit/pull/66/commits/9a309369d3659a19c4d00c6e345ed6954498b666)

**Comment by w11th** [comment]
> Also, at what point should we switch to OTEL, with our own exporter?

@bodhi I think we can start the switch now. I can start with `mcd-ketchup`.

OTEL is well-known and full-featured. If we decide to switch to OTEL, it seems unnecessary to keep adding features (remote tracing/adding events...) to `logtracing`. 
(Jaden and I talked about this a while ago, before this [PR](https://github.com/theplant/appkit/pull/68).)

File: logtracing/README.md [hunk 1]
```diff
@@ -140,10 +140,49 @@ For the server requests:
...
+1. Implement the `Exporter` interface:
+    ```Go
+    type Exporter interface {
+    	ExportSpan(s *SpanData)
+    }
+    ```
+2. Initialize and register your exporter:
+    ```Go
+    exporter := <your exporter>
+    logtracing.RegisterExporter(expoter)
```
Comment (bodhi) [inline logtracing/README.md:156]:
```suggestion
    logtracing.RegisterExporter(exporter)
```

File: logtracing/README.md [hunk 2]
```diff
@@ -140,10 +140,49 @@ For the server requests:
...
+    ```
+
+When you register exporters, `logtracing` will pass sampled spans to the exporters after logging.
```
Comment (bodhi) [inline logtracing/README.md:159]: Ah, so this will log *and* export to Honeycomb? I thought it was a replacement!

**Review by bodhi — APPROVED** [review]
Thanks @w11th

**Comment by bodhi** [comment]
> I think we can start the switch now. I can start with mcd-ketchup.

Can you do a PR on a single component, so we can see the difference?

**Comment by w11th** [comment]
> Can you do a PR on a single component, so we can see the difference?

YES, I will do that. https://theplanttokyo.atlassian.net/browse/MKP-249

**Pull request merged**